syncstorage/storage/memcached.py

```diff
@@ -944,23 +945,31 @@ def del_item(self, userid, item):
             raise ItemNotFoundError
         return modified
 
-    def get_cached_batches(self, userid):
-        return self.cache.gets(self.get_batches_key(userid))
+    def get_cached_batches(self, userid, ts=None):
+        if ts is None:
+            ts = get_timestamp()
+        ts = int(ts)
+        bdata, bcasid = self.cache.gets(self.get_batches_key(userid))
+        # Remove any expired batches, but let the
+        # calling code write it back out to memcache.
+        if bdata:
+            for batchid, batch in bdata.items():
+                if batch["created"] + BATCH_LIFETIME < ts:
+                    del bdata[batchid]
+        return bdata, bcasid
 
     def create_batch(self, userid):
-        bdata, bcasid = self.get_cached_batches(userid)
-        batch = get_timestamp()
-        batchid = int(batch * 1000)
+        ts = get_timestamp()
+        bdata, bcasid = self.get_cached_batches(userid, ts)
+        batchid = int(ts * 1000)
         if not bdata:
             bdata = {}
         if batchid in bdata:
             raise ConflictError
-        bdata[batchid] = {"user": userid,
-                          "modified": batch,
-                          # FIXME Rough guesstimate of the maximum
-                          #       reasonable life span of a batch
-                          "expires": int(batch) + 2 * 3600,
-                          "items": []}
+        bdata[batchid] = {
+            "created": int(ts),
+            "items": []
+        }
```

Comment:
The "user" and "modified" keys were never used, so I removed them to save space.  Also changed "expires" to "created" in case we want to make the batch lifetime configurable in future.